Formula/emqx.rb

```diff
@@ -0,0 +1,45 @@
+class Emqx < Formula
+  desc "MQTT broker for IoT"
+  homepage "https://www.emqx.io/"
+  url "https://github.com/emqx/emqx/archive/refs/tags/v5.0.11.tar.gz"
+  sha256 "5b797f768961abccf9ea0050613c02b7d1fc1d3877306d592e72c64a8d95588b"
+  license "Apache-2.0"
+  head "https://github.com/emqx/emqx.git", branch: "master"
+
+  depends_on "autoconf"  => :build
+  depends_on "automake"  => :build
+  depends_on "cmake"     => :build
+  depends_on "coreutils" => :build
+  depends_on "erlang@24" => :build
+  depends_on "freetds"   => :build
+  depends_on "libtool"   => :build
+  depends_on "openssl@3"
+
+  uses_from_macos "curl"    => :build
+  uses_from_macos "unzip"   => :build
+  uses_from_macos "zip"     => :build
+
+  on_linux do
+    depends_on "ncurses"
+    depends_on "zlib"
+  end
```

Comment:
This is better
```suggestion
  uses_from_macos "ncurses"
  uses_from_macos "zlib"
```

Reply:
Would you like me to fix this now, or can I do it when I update the formula?

Reply:
Let's update it, I have already tested out on a linux box.

Reply:
fixed

Reply:
@chenrui333 linkage test on linux failed after this change

Reply:
@chenrui333 how would you advise to proceed?

Reply:
it should be

```
  uses_from_macos "ncurses"
  uses_from_macos "zlib"
```

I have to dig a bit myself.

Reply:
Maybe let's just use `depends_on` for the formula merge? cc @id